packages/elements/src/react/utils/create-context-for-dom-validation.ts

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+
+/**
+ * Use this context helper to detect whether a component has a particular parent higher up in the DOM or not.
+ */
+export function createContextForDomValidation(displayName: string) {
+  const ReactContext = React.createContext(false);
+  const OriginalProvider = ReactContext.Provider;
+
+  function Provider({ children }: WithChildrenProp) {
+    return React.createElement(
+      OriginalProvider,
+      {
+        value: true,
+      },
+      children,
+    );
+  }
+
+  Provider.displayName = displayName;
+
+  function useContext(allowMissingContext: boolean = false) {
```

Comment:
The helper is very much a copy of https://github.com/clerk/javascript/blob/f984829a70fbeae0d6faef58248ddd368a771cce/packages/elements/src/react/utils/create-context-from-actor-ref.ts so that's why it's here. I think we can keep it and then at least have a fallback error if we don't handle it ourselves